xbmc/video/VideoDatabase.cpp

```diff
@@ -9572,7 +9604,9 @@ void CVideoDatabase::SplitPath(const std::string& strFileNameAndPath, std::strin
   else if (URIUtils::IsPlugin(strFileNameAndPath))
   {
     CURL url(strFileNameAndPath);
-    strPath = url.GetWithoutFilename();
+    strPath = url.GetWithoutOptions();
+    if (strPath == strFileNameAndPath)
+      strPath = url.GetWithoutFilename();
```

Comment:
@AkariDN This change seems to have broken mark as watched for some add-ons. See https://trac.kodi.tv/ticket/17947 for reference.